src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
I don't understand what you guys want. I find it really, really critical to mention the reasoning for the deprecation, because it is two-fold: The only (!) reason "newFromArray" exists in the first place is because of this one line in the DataValueDeserializer. There was no alternative for this a while ago, but is now. So why not mention this alternative? And the name is just bad, which caused the current pain in the first place. I was under the impression a method that is literally called "newFromArray" always gets an array. But this is wrong. Users should introduce a new one with a better name, if they really need one.